Services/housekeeping_service.c

```diff
@@ -55,7 +55,26 @@ SAT_returnState hk_service_app(csp_packet_t *packet) {
 }
 
 static SAT_returnState hk_parameter_report(csp_packet_t *packet) {
-  HAL_hk_report(packet->data[SID_byte], packet->data + SID_byte + 1);
+  if(HAL_hk_report(packet->data[SID_byte], packet->data + SID_byte + 1) != SATR_OK) {
```

Comment:
Ok I'm beginning to rethink the re-use of the packet. I'm rethinking because the size of the data feild of the incoming packet is only as long as the amount of data recieved (which will likely be much smaller than the data we want to send down). Take a look at this [doc](https://github.com/libcsp/libcsp/blob/87006959696c78f70535ab382b0bcd4cb5a6558d/doc/basic.rst)
I think it might be best to use a new CSP packet. I'm thinking
```
static SAT_returnState hk_parameter_report(csp_packet_t *packet) {
  csp_packet_t return_packet = csp_buffer_new(32); <- the parameter is the size of the data feild. We should choose a number that will accomodate for the largest housekeeping value (for now, just choose something)
  return_packet->data[SID_byte] = packet->data[SID_byte];
  ...
  HAL_hk_report(packet->data[SID_byte], return_packet->data + SID_byte + 1); < - load the data into the new packet
  return_packet_header(return_packet, packet); <-- modify the return function to copy header data from one packet to another
  csp_buffer_free(packet); <- only free if all the other steps succeeded...
  queue_packet(return_packet);

Reply:
NEVERMIND, scratch that. I found out that *actually* the value passed to `csp_buffer_get` doesn't actually effect the size of the data feild. Rather it's just used to make sure you're not requesting more than the configured limit.
Check out the implementation https://github.com/libcsp/libcsp/blob/81154495f62ba85928e5350497314e64547aa9ed/src/csp_buffer.c#L109

So actually, we can just keep doing it this way. But yeah, do the memory size check before we load the data into the buffer!